doc/crypto/threads.pod

```diff
@@ -2,176 +2,63 @@
 
 =head1 NAME
 
-CRYPTO_THREADID_set_callback, CRYPTO_THREADID_get_callback,
-CRYPTO_THREADID_current, CRYPTO_THREADID_cmp, CRYPTO_THREADID_cpy,
-CRYPTO_THREADID_hash, CRYPTO_set_locking_callback, CRYPTO_num_locks,
-CRYPTO_set_dynlock_create_callback, CRYPTO_set_dynlock_lock_callback,
-CRYPTO_set_dynlock_destroy_callback, CRYPTO_get_new_dynlockid,
-CRYPTO_destroy_dynlockid, CRYPTO_lock - OpenSSL thread support
```

Comment:
The old API is really ugly IMO, and it should probably be removed completely.

Could we move all the old APIs to an internal header so they wouldn't be exported anymore (even though internally they would still be used, at least until the conversion is finished)? This way the 1.1 API is already going to be clean (even if the conversion hasn't finished yet) and we won't have to break the API again in the future.

I'd also prioritize converting locks that are exposed to users via e.g. structures and similar before 1.1 is finalized.